string data
int64 crc32